packages/transport-test/e2e/api/api.test.ts

```diff
@@ -0,0 +1,118 @@
+    const apiTestFixture: ApiTestFixture[] = [];
+    if (typeof window !== 'undefined') {
+        apiTestFixture.push({ initArgs: { usbInterface }, description: 'browser' });
+    } else {
+        apiTestFixture.push(
+            {
+                initArgs: { usbInterface, forceReadSerialOnConnect: true },
+                description: 'node, forceReadSerialOnConnect: true',
+            },
+            {
+                initArgs: { usbInterface, forceReadSerialOnConnect: false },
+                description: 'node, forceReadSerialOnConnect: false',
+            },
+        );
+    }
+
+    return apiTestFixture.map(f => {
+        return { api: new UsbApi(f.initArgs), description: f.description };
+    });
+};
+
+const runTests = async () => {
+    const fixtures = await setupApisUnderTest();
+
+    for (const f of fixtures) {
+        info(`Running tests for ${f.description}`);
+        const api = f.api;
+        let path: string;
+
+        const getConnectedDevicePath = async () => {
+            info('getConnectedDevicePath...');
+            const res = await api.enumerate();
+            debug('getConnectedDevicePath: discovered devices', res);
+
+            assertSuccess(res);
+            if (res.payload.length !== 1) {
+                throw new Error(error('Expected exactly one device to be connected'));
+            }
+            debug('getConnectedDevicePath: path set to: ', res.payload[0].path);
+            path = res.payload[0].path;
+        };
+
+        const pingPong = async () => {
+            await api.openDevice(path, true);
+            const writeResponse = await api.write(path, buildMessage('PING'));
+            debug('writeResponse', writeResponse);
+            assertSuccess(writeResponse);
+
+            const readResponse = await api.read(path);
+
+            debug('readResponse', readResponse);
+            assertSuccess(readResponse);
+            debug('readResponse', readResponse.payload.toString('hex'));
+            assertMessage(readResponse.payload, 'SUCCESS');
+            await api.closeDevice(path);
+        };
+
+        const concurrentEnumerate = async (concurrent: number) => {
+            const res = await Promise.all(
+                new Array(concurrent).fill(undefined).map(() => api.enumerate()),
+            );
+            res.forEach((r, index) => {
+                assertSuccess(r);
+                assertSuccess(res[0]);
+                if (index > 0) {
+                    assertEquals(r.payload, res[0].payload);
+                }
+            });
+        };
+
+        // interestingly, in node, concurrent enumeration does not work if triggered as the first interaction with connected device.
+        // concurrent enumeration triggered couple of lines below works correctly
+        await sharedTest('concurrent enumerate - as the first operation', () =>
+            // todo: 3 doesn't work in node!!! FIX
+            concurrentEnumerate(typeof window === 'undefined' ? 3 : 1),
```

Comment:
one way how to fix this https://github.com/trezor/trezor-suite/compare/transport-synchronize-enumeration?expand=1